maas/data_source_maas_package_repositories.go

```diff
@@ -0,0 +1,105 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceMAASPackageRepositories() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMAASPackageRepositoriesRead,
+		Description: "Provides a data source to fetch MAAS package repositories.",
+
+		Schema: map[string]*schema.Schema{
+			"arches": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The list of supported architectures.",
+			},
+			"components": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The list of components to enable. Only applicable to custom repositories.",
```

Comment:
Nit: as this is a data source, wording tweak
```suggestion
				Description: "The list of components that are enabled. Only applicable to custom repositories.",
```
```suggestion
				Description: "The list of components to enable. Only applicable to custom repositories.",
```